setup/cli-install.php

```diff
@@ -0,0 +1,183 @@
+        'default' => dirname($path) . '/connectors/',
+    ],
+    'context_connectors_url' => [
+        'prompt' => 'context_connector_url',
+        'default' => '/connectors/',
+    ],
+    'remove_setup_directory' => 1,
+];
+
+// Parse CLI arguments
+$args = array_slice($argv, 1);
+$cli_variables = [];
+foreach ($args as $arg) {
+    $tmp = array_map('trim', explode('=', $arg));
+    if (count($tmp) === 2) {
+        $k = ltrim($tmp[0], '-');
+        if (isset($variables[$k])) {
+            $cli_variables[$k] = $tmp[1];
+        } elseif ($k == 'mode' && in_array($tmp[1], ['new', 'upgrade', 'advanced'])) {
+            $mode = $tmp[1];
+        }
+    }
+}
+
+$lang = !empty($cli_variables['language']) && in_array($cli_variables['language'], $languages)
+    ? $cli_variables['language']
+    : 'en';
+/** @var array $_lang */
+require($path . 'lang/' . $lang . '/default.inc.php');
+
+// If it is upgrade - parse old variables
+$old_variables = [];
+if (file_exists($config)) {
+    $use = ($mode == 'upgrade')
+        ? 'Y'
+        : readline("There is an existing config file found at {$config}. Do you want to load this values? (Y): ");
+    if (empty($use) || strtolower($use) == 'y') {
+        $tmp = simplexml_load_file($config);
+        $tmp = @json_decode(json_encode($tmp), true);
+        if (is_array($tmp)) {
+            $old_variables = $tmp;
+        }
+    }
+}
+
+// Merge everything
+$variables = array_merge($variables, $old_variables, $cli_variables);
+
+// Process config
+$data = [];
+foreach ($variables as $key => $params) {
+    if (!is_array($params)) {
+        $data[$key] = $params;
+    } elseif (!empty($params['prompt'])) {
+        $prompt = rtrim($_lang[$params['prompt']], ':');
+        $res = '';
+        if (!empty($params['default'])) {
+            if ($key == 'language') {
+                $prompt .= ' (' . implode('|', $languages) . ') ';
+            }
+            $prompt .= " ({$params['default']})";
+            if (!$res = readline($prompt . ': ')) {
+                $res = $params['default'];
+            }
+        } else {
+            while (!$res = readline($prompt . ': ')) {
+                // pass
+            }
+            readline_add_history(trim($res));
+        }
+        $data[$key] = trim($res);
+    }
+    if (strpos($key, '_path') !== false || strpos($key, '_url') !== false) {
+        $data[$key] = preg_replace('#/+#', '/', ('/' . trim($data[$key], '/') . '/'));
+    }
+}
+
+// Check directories
+$dirs = [
+    'core_path' => '/core/',
+    'context_mgr_path' => '/manager/',
+    'context_connectors_path' => '/connectors/',
+];
+foreach ($dirs as $key => $value) {
+    $target = dirname($path) . $value;
+    if (file_exists($target) && $data[$key] != $target && strpos($data[$key], dirname($path)) === 0) {
+        shell_exec('rm -rf ' . escapeshellarg($data[$key]));
+        shell_exec('mv ' . escapeshellarg($target) . ' ' . escapeshellarg($data[$key]));
+    }
+}
```

Comment:
I'm not sure this is the best idea and is not compatible across all supported platforms. Would it not be better to use native PHP features to handle this?

Reply:
@opengeek You are right, I`m totally forgot about Windows.

Will try to rewrite it tomorrow.